library_generation/utils/generation_config_comparator.py

```diff
@@ -38,10 +38,12 @@ def compare_config(
     :return: a ConfigChange objects.
     """
     diff = defaultdict(list[LibraryChange])
+    # these params are not used in config comparison.
+    excluded_params = {"path_to_yaml"}
```

Comment:
Why do we need this parameter in `GenerationConfig` class? How is it used? I thought `GenerationConfig` is a pure representation for the yaml, but looks like it includes more things.

Reply:
This parameter is added in #2443.

Basically, this parameter is used to run template exclusion using our own implementation of synthtool.